### ISSUE TITLE
Setting Proxy Env Variable to ""

### ISSUE THREAD

**Issue by tacchino** [test-kitchen/test-kitchen#934]
Since local proxy environment variables are automatically passed in to the test machine we need a solution to support test environments that should have not proxy environment variables set. Explicitly setting proxy to "" as suggested in https://github.com/test-kitchen/test-kitchen/pull/813 will fail in cases where the libraries using those variables don't evaluate the value prior to operating on them.

Example:

```
Recipe: sensu::_linux
         * apt_repository[sensu] action add
           * remote_file[/tmp/kitchen/cache/pubkey.gpg] action create

             ================================================================================
             Error executing action `create` on resource 'remote_file[/tmp/kitchen/cache/pubkey.gpg]'
             ================================================================================

             NoMethodError
             -------------
             undefined method `host' for "":String

             Resource Declaration:
             ---------------------
             # In /tmp/kitchen/cache/cookbooks/apt/providers/repository.rb

       90:     remote_file cached_keyfile do
       91:       source new_resource.key
       92:       mode 00644
       93:       sensitive new_resource.sensitive if respond_to?(:sensitive)
       94:       action :create
       95:     end
       96:   else

             Compiled Resource:
             ------------------
             # Declared in /tmp/kitchen/cache/cookbooks/apt/providers/repository.rb:90:in `install_key_from_uri'

             remote_file("/tmp/kitchen/cache/pubkey.gpg") do
        provider Chef::Provider::RemoteFile
        action [:create]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        path "/tmp/kitchen/cache/pubkey.gpg"
        backup 5
        atomic_update true
        source ["http://repositories.sensuapp.org/apt/pubkey.gpg"]
        use_etag true
        use_last_modified true
        declared_type :remote_file
        cookbook_name "sensu"
        mode 420
             end
```

**Comment by mwrock** [comment]
yeah better not to set or set to `$null`

**Comment by lcc2207** [comment]
Im seeing the same issue, is the only fix for now to uninstall 1.5 and fail back to an older version?
